src/decorators/getValidators.integration.test.ts

```diff
@@ -0,0 +1,75 @@
+import { describe, it, expect } from 'vitest';
+import { Address, createPublicClient, http } from 'viem';
+import { generatePrivateKey, privateKeyToAccount } from 'viem/accounts';
+
+import { nitroTestnodeL2 } from '../chains';
+import { rollupAdminLogicPublicActions } from './rollupAdminLogicPublicActions';
+import { getInformationFromTestnode, getNitroTestnodePrivateKeyAccounts } from '../testHelpers';
+import { getValidators } from '../getValidators';
+
+const { l3RollupOwner } = getNitroTestnodePrivateKeyAccounts();
+const { l3Rollup, l3UpgradeExecutor } = getInformationFromTestnode();
+
+const client = createPublicClient({
+  chain: nitroTestnodeL2,
+  transport: http(),
+}).extend(
+  rollupAdminLogicPublicActions({
+    rollup: l3Rollup,
+  }),
+);
+
+async function toggleValidator(validator: Address, state: boolean) {
+  const tx = await client.rollupAdminLogicPrepareTransactionRequest({
+    functionName: 'setValidator',
+    args: [[validator], [state]],
+    account: l3RollupOwner.address,
+    upgradeExecutor: l3UpgradeExecutor,
+    rollup: l3Rollup,
+  });
+
+  await client.sendRawTransaction({
+    serializedTransaction: await l3RollupOwner.signTransaction(tx),
+  });
+}
+
+describe('successfully get validators', async () => {
+  it('when disabling the same validator multiple time', async () => {
+    const randomAccount = privateKeyToAccount(generatePrivateKey()).address;
+
+    const initialValidators = await getValidators(client, { rollupAddress: l3Rollup });
+    // By default, nitro chains have 10 validators
+    expect(initialValidators).toHaveLength(10);
+
+    await toggleValidator(randomAccount, false);
+    await toggleValidator(randomAccount, false);
+
+    const newValidators = await getValidators(client, { rollupAddress: l3Rollup });
+    // Setting the same validator multiple time to false doesn't add new validators
+    expect(newValidators).toEqual(initialValidators);
+
+    await toggleValidator(randomAccount, true);
```

Comment:
this test shouldn't involve adding a validator after removing the validator

on another note, I'd also prefer breaking down this removal test to a few tests

1. remove an existing validator once
2. remove an existing validator twice
3. remove an address that is not a validator once
4. remove an address that is not a validator twice
5. add a validator and remove it

Reply:
The test is mostly here to test case when you remove n times, add once at the end, to be sure it's considered added. Same for added multiple time, then removed once.
I don't think we would benefit from having both 1 and 2, or 3 and 4.

Reply:
i think performing an action (add/remove) with an existing validator vs a non-validator address is different so it'd be good to have both, but agreed that we can skip once vs twice cases